python/lsst/ap/association/dia_collection.py

```diff
@@ -19,3 +19,188 @@
 # the GNU General Public License along with this program.  If not,
 # see <http://www.lsstcorp.org/LegalNotices/>.
 #
+
+from __future__ import absolute_import, division, print_function
+
+from scipy.spatial import cKDtree
+
+import .dia_object
+
+
+class DIAObjectCollection(object):
+    """ A collection of DIAObjects with convenience functions for scoring and
+    matching DIASources into the collection of DIAObjects.
+
+    Attributes
+    ----------
+    dia_objects : a list of DIAObjects
+        List of DIAObjects representing this collection the current (e.g.)
```

Comment:
That makes sense. One hesitation I have for this is that the scoring and matching step is most easily done on the relative position in a list rather than a id in a dictionary. What if I add a dictionary internal to the class that is keyed on the id and returns the index in the list? I think I would hide this from user through some overwriting of built in functions or create a accessor method that uses the id as the input and returns the DIAObject of interest through the dictionary of id to list position.